lib/dal/src/change_set.rs

```diff
@@ -252,14 +271,45 @@ impl ChangeSet {
         Ok(change_set)
     }
 
-    // pub fn generate_ulid(&self) -> ChangeSetResult<Ulid> {
-    //     self.generator
-    //         .lock()
-    //         .map_err(|e| ChangeSetError::Mutex(e.to_string()))?
-    //         .generate()
-    //         .map(Into::into)
-    //         .map_err(Into::into)
-    // }
+    pub async fn into_frontend_type(
+        &self,
+        ctx: &DalContext,
+    ) -> ChangeSetResult<si_frontend_types::ChangeSet> {
+        let merge_requested_by_email =
+            if let Some(merge_requested_by) = self.merge_requested_by_user_id {
+                User::get_by_pk(ctx, merge_requested_by)
+                    .await?
+                    .map(|user| user.email().clone())
+            } else {
+                None
+            };
+
+        let reviewed_by_email = if let Some(reviewed_by) = self.reviewed_by_user_id {
+            User::get_by_pk(ctx, reviewed_by)
+                .await?
+                .map(|user| user.email().clone())
+        } else {
+            None
+        };
+
+        let change_set = si_frontend_types::ChangeSet {
+            created_at: self.created_at,
+            id: self.id.into(),
+            updated_at: self.updated_at,
+            name: self.name.clone(),
+            status: self.status.into(),
+            base_change_set_id: self.base_change_set_id.map(|id| id.into()),
+            workspace_id: self.workspace_id.map_or("".to_owned(), |id| id.to_string()),
+            merge_requested_by_user_id: self.merge_requested_by_user_id.map(|s| s.to_string()),
+            merge_requested_by_user_email: merge_requested_by_email,
+            merge_requested_at: self.merge_requested_at,
+            reviewed_by_user_id: self.reviewed_by_user_id.map(|id| id.into()),
+            reviewed_by_user_email: reviewed_by_email,
+            reviewed_at: self.reviewed_at,
+        };
+
+        Ok(change_set)
+    }
 
     pub async fn update_workspace_id(
```

Comment:
This method primarily updates the `status` of the change set? Not important for this PR but maybe we should consider a rename if that's its purpose.